iceoryx_hoofs/CMakeLists.txt

```diff
@@ -230,10 +147,20 @@ target_compile_options(iceoryx_hoofs PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANIT
 # TODO: Make ICEORYX::UTILS private???
 target_include_directories(iceoryx_hoofs
     PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include/${PREFIX}>
 )
 
+#
+########## hoofs testing ##########
+#
+
+if(BUILD_TEST)
+    find_package(GTest CONFIG REQUIRED)
+    add_subdirectory(testing)
+    add_subdirectory(test)
+endif(BUILD_TEST)
```

Comment:
This changes the semantics for the testing library. Before it was build when gtest was found, now only when the tests are build. Le's assume someone integrates icoryx as subrepo in a bigger code-base and wants to use the testing lib then it is not anymore possible without also building all the iceoryx tests.

@dkroenke do we want to keep being able to build the testing lib without building the tests?

Reply:
Just to confirm my understanding here, before we had the following snippet

``` cmake
if(BUILD_TEST AND NOT GTest_FOUND)
    find_package(GTest CONFIG REQUIRED)
endif(BUILD_TEST AND NOT GTest_FOUND)
```

which meant that gtest was only looked for if `BUILD_TEST` was `ON`. That means that `testing` was only built if the tests were being built.

Are you saying that if you for example do `add_subdirectory(iceoryx/iceoryx_hoofs)` a `find_package(GTest REQUIRED)` call from the parent project will transfer to this one and enable `testing`?

for example
```cmake
project(parent_proj)

find_package(GTest REQUIRED)

add_subdirectory(iceoryx/iceoryx_hoofs)  # <<--- Will this call enable iceoryx_hoofs_testing??
```

Reply:
@andre-nguyen yes, that would enable the `testing` lib. In case someone is building a framework on top of iceoryx, this might be desirable. One could for example use a released version of iceoryx without increasing the build time by also building the tests. I'm just not sure if we want to solve the problem this way. Maybe a dedicated cmake switch would make more sense if someone really needs use iceoryx the way I described it.

My tendency is to simplify it like you did and introduce this when someone needs it but let's see if @dkroenke agrees.

Reply:
@elBoberido @andre-nguyen 

Normally the `testing` functions are tightly coupled the the iceoryx tests. I see there generic functions like the timing test stubs but on the other hand quite iceoryx -specific mocks like a mock for the error handler.

Then we could do something like this:
```cmake
if(NOT GTest_FOUND AND BUILD_TEST)
    find_package(GTest CONFIG REQUIRED)
elseif(GTest_FOUND)
    add_subdirectory(testing)
elseif(GTest_FOUND AND BUILD_TEST)
   add_subdirectory(test)
endif()
```

This snippet is untested, this could work maybe, @andre-nguyen can you test that?
The intention is not to save codelines but have all the test-related logic collected a bit into a section in cmake.

Reply:
@dkroenke that code snippet won't work. It must be something like
```cmake
# finding gtest and adding the subdirectories is split to support the use case of
# building the testing lib without the tests by providing gtest externally
if(NOT GTest_FOUND AND BUILD_TEST)
    find_package(GTest CONFIG REQUIRED)
endif()

if(GTest_FOUND)
    add_subdirectory(testing)
    if(BUILD_TEST)
        add_subdirectory(test)
    endif()
endif()
```